pkg/test/framework/components/istio/ingress.go

```diff
@@ -221,7 +242,9 @@ func (c *ingressImpl) callEcho(opts echo.CallOptions) (echo.CallResult, error) {
 		opts.HTTP.Headers.Set(headers.Host, host)
 	}
 	// Default address based on port
-	opts.Address = addr
+	if opts.Address == "" {
+		opts.Address = addr
```

Comment:
by default callEcho would overwrite the address with the first remote address that is fine.
This would be bad in case someone would use this function for DS, hence I removed the overwrite and I allow the caller to specificy one address the want to test via the options